flexmeasures/api/v3_0/sensors.py

```diff
@@ -552,21 +552,25 @@ def get_schedule(self, sensor: Sensor, job_id: str, duration: timedelta, **kwarg
             return unknown_schedule("Scheduling job has an unknown status.")
         schedule_start = job.kwargs["start"]
 
-        schedule_data_source_name = "Seita"
-        if "data_source_name" in job.meta:
-            schedule_data_source_name = job.meta["data_source_name"]
-        scheduler_source = DataSource.query.filter_by(
-            name=schedule_data_source_name, type="scheduling script"
-        ).one_or_none()
-        if scheduler_source is None:
+        data_source_info = job.meta.get("data_source_info")
+        if data_source_info is None:
+            data_source_info = dict(
+                name="Seita"
+            )  # TODO: change to raise later - all scheduling jobs now get full info
+        scheduler_sources = DataSource.query.filter_by(
+            type="scheduling script",
+            **data_source_info,
+        ).all()  # there can be more than one, e.g. different users
+        if len(scheduler_sources) == 0:
+            s_info = ",".join([f"{k}={v}" for k, v in data_source_info.items()])
             return unknown_schedule(
-                error_message + f'no data is known from "{schedule_data_source_name}".'
+                error_message + f"no data is known from [{s_info}]."
             )
```

Comment:
This code looks duplicated from `v1_3/implementations.py`. Could we refactor it to a util function?

Reply:
Yes, I can refactor this into a util function `get_data_source_fpr_job.